hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/FileSizeCountTask.java

```diff
@@ -128,31 +128,39 @@ public Pair<String, Boolean> process(OMUpdateEventBatch events) {
         continue;
       }
       String updatedKey = omdbUpdateEvent.getKey();
-      OmKeyInfo omKeyInfo = omdbUpdateEvent.getValue();
+      Object value = omdbUpdateEvent.getValue();
 
-      try {
-        switch (omdbUpdateEvent.getAction()) {
-        case PUT:
-          handlePutKeyEvent(omKeyInfo, fileSizeCountMap);
-          break;
+      if (value instanceof OmKeyInfo) {
+        OmKeyInfo omKeyInfo = (OmKeyInfo) value;
 
-        case DELETE:
-          handleDeleteKeyEvent(updatedKey, omKeyInfo, fileSizeCountMap);
-          break;
+        try {
+          switch (omdbUpdateEvent.getAction()) {
+          case PUT:
+            handlePutKeyEvent(omKeyInfo, fileSizeCountMap);
+            break;
 
-        case UPDATE:
-          handleDeleteKeyEvent(updatedKey, omdbUpdateEvent.getOldValue(),
-              fileSizeCountMap);
-          handlePutKeyEvent(omKeyInfo, fileSizeCountMap);
-          break;
+          case DELETE:
+            handleDeleteKeyEvent(updatedKey, omKeyInfo, fileSizeCountMap);
+            break;
 
-        default: LOG.trace("Skipping DB update event : {}",
-            omdbUpdateEvent.getAction());
+          case UPDATE:
+            handleDeleteKeyEvent(updatedKey, omdbUpdateEvent.getOldValue(),
+                fileSizeCountMap);
+            handlePutKeyEvent(omKeyInfo, fileSizeCountMap);
+            break;
+
+          default:
+            LOG.trace("Skipping DB update event : {}",
+                omdbUpdateEvent.getAction());
+          }
+        } catch (Exception e) {
+          LOG.error("Unexpected exception while processing key {}.",
+              updatedKey, e);
+          return new ImmutablePair<>(getTaskName(), false);
         }
-      } catch (Exception e) {
-        LOG.error("Unexpected exception while processing key {}.",
-                updatedKey, e);
-        return new ImmutablePair<>(getTaskName(), false);
+      } else {
+        LOG.warn("Unexpected value type {} for key {}. Skipping processing.",
```

Comment:
If we believe it's common case to have RepeatedOmKeyInfo in future considering the hsync implementation, debug level will be a appropriate log level for this message.

Reply:
Thanks for the suggestion @ChenSammi I have made the changes!